src/PrestaShopBundle/EventListener/AccessDeniedListener.php

```diff
@@ -104,7 +105,9 @@ private function getAccessDeniedResponse(Request $request, AdminSecurity $adminS
             ], Response::HTTP_FORBIDDEN);
         }
 
-        $this->session->getFlashBag()->add('error', $this->getErrorMessage($adminSecurity));
+        if ($this->session instanceof Session) {
```

Comment:
I'm still not sure we should have this kind of check, if the SessionInterface is not the right class to use we should change it for Session instead :thinking:

Reply:
![image](https://user-images.githubusercontent.com/1533248/88902988-3c0fac00-d253-11ea-941b-8b0499a1fea5.png)

Reply:
This looks good enough to me 😊

Reply:
@Progi1984 :sweat_smile: We should always prefer interface when it's possible.
Here, we are using methods not register in the Interface, and the whole code used by the `$session` is in a condition to check if the class is an instance of Session. The interface has no aim here :thinking:

Reply:
Changed the signature. Documented the BC.